src/basic/stdio-util.h

```diff
@@ -9,13 +9,19 @@
 #include "macro.h"
 #include "memory-util.h"
 
-#define snprintf_ok(buf, len, fmt, ...)                                \
-        ({                                                             \
-                char *_buf = (buf);                                    \
-                size_t _len = (len);                                   \
-                int _snpf = snprintf(_buf, _len, (fmt), __VA_ARGS__);  \
-                _snpf >= 0 && (size_t) _snpf < _len ? _buf : NULL;     \
-        })
+_printf_(3, 4)
+static inline char *snprintf_ok(char *buf, size_t len, const char *format, ...) {
+        va_list ap;
+        int r;
+
+        va_start(ap, format);
```

Comment:
I want to pass a pre-allocated buffer via a compound initializer. If we have the macro + statement expressions, the compound initializer block will be scoped to the macro block. If we make it a static inline function, it will be scoped to the block of the function calling snprintf_ok().

Reply:
Can you add this rationale to the commit description?

>  hmm, i wonder how the generated code here actually compares to the status quo ante?

That's a good question, but since we only have 30 invocations of snprintf_ok in the codebase, we don't really care if it's particularly efficient.